src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -45,7 +47,36 @@ class UserService(
     return userRepository.findAll(hasQualificationsAndRoles(qualifications, roles), Sort.by(Sort.Direction.ASC, "name"))
   }
 
-  fun getUserForAllocation(qualifications: List<UserQualification>): UserEntity? = userRepository.findQualifiedAssessorWithLeastPendingAllocations(qualifications.map(UserQualification::toString), qualifications.size.toLong())
+  fun getUserForAssessmentAllocation(application: ApplicationEntity): UserEntity {
+    val unsuitableUsers = mutableListOf<UUID>(UUID.randomUUID())
+    val qualifications = application.getRequiredQualifications()
+
+    while (true) {
+      val potentialUser = userRepository.findQualifiedAssessorWithLeastPendingAssessments(qualifications.map(UserQualification::toString), qualifications.size.toLong(), unsuitableUsers)
+        ?: throw RuntimeException("No Users with all of required qualifications (${qualifications.joinToString(", ")}) that also pass LAO could be found")
+
+      if (offenderService.getOffenderByCrn(application.crn, potentialUser.deliusUsername) !is AuthorisableActionResult.Success) {
+        unsuitableUsers += potentialUser.id
+      } else {
+        return potentialUser
+      }
+    }
+  }
+
+  fun getUserForPlacementApplicationAllocation(crn: String): UserEntity {
+    val unsuitableUsers = mutableListOf<UUID>(UUID.randomUUID())
+
+    while (true) {
+      val potentialUser = userRepository.findQualifiedMatcherWithLeastPendingPlacementApplications(unsuitableUsers)
+        ?: throw RuntimeException("No Users that pass LAO could be found")
+
+      if (offenderService.getOffenderByCrn(crn, potentialUser.deliusUsername) !is AuthorisableActionResult.Success) {
```

Comment:
There's potentially going to be a lot of upstream calls here, not the end of the world given we're racing to a deadline, but might be worth putting a `TODO` comment here to figure out a tidier way of getting users who can access an LAO?